contracts-js/BTCUtils.js

```diff
@@ -0,0 +1,1012 @@
+//     }
+
+  /// @notice      Checks that the vin passed up is properly formatted
+  /// @dev         Consider a vin with a valid vout in its scriptsig
+  /// @param _vout Raw bytes length-prefixed output vector
+  /// @return      True if it represents a validly formatted bout
+  validateVout: (vout) => {
+    return
+  },
+
+
+
+  /* ************ */
+  /* Block Header */
+  /* ************ */
+
+//     /// @notice          Extracts the transaction merkle root from a block header
+//     /// @dev             Use verifyHash256Merkle to verify proofs with this root
+//     /// @param _header   The header
+//     /// @return          The merkle root (little-endian)
+//     function extractMerkleRootLE(bytes memory _header) internal pure returns (bytes memory) {
+//         return _header.slice(36, 32);
+//     }
+
+  /// @notice          Extracts the transaction merkle root from a block header
+  /// @dev             Use verifyHash256Merkle to verify proofs with this root
+  /// @param _header   The header
+  /// @return          The merkle root (little-endian)
+  extractMerkleRootLE: (header) => {
+    return
+  },
+
+//     /// @notice          Extracts the transaction merkle root from a block header
+//     /// @dev             Use verifyHash256Merkle to verify proofs with this root
+//     /// @param _header   The header
+//     /// @return          The merkle root (big-endian)
+//     function extractMerkleRootBE(bytes memory _header) internal pure returns (bytes memory) {
+//         return reverseEndianness(extractMerkleRootLE(_header));
+//     }
+
+  /// @notice          Extracts the transaction merkle root from a block header
+  /// @dev             Use verifyHash256Merkle to verify proofs with this root
+  /// @param _header   The header
+  /// @return          The merkle root (big-endian)
+  extractMerkleRootBE: (header) => {
+    return
+  },
+
+//     /// @notice          Extracts the target from a block header
+//     /// @dev             Target is a 256 bit number encoded as a 3-byte mantissa and 1 byte exponent
+//     /// @param _header   The header
+//     /// @return          The target threshold
+//     function extractTarget(bytes memory _header) internal pure returns (uint256) {
+//         bytes memory _m = _header.slice(72, 3);
+//         uint8 _e = uint8(_header[75]);
+//         uint256 _mantissa = bytesToUint(reverseEndianness(_m));
+//         uint _exponent = _e - 3;
+
+//         return _mantissa * (256 ** _exponent);
+//     }
+
+  /// @notice          Extracts the target from a block header
+  /// @dev             Target is a 256 bit number encoded as a 3-byte mantissa and 1 byte exponent
+  /// @param _header   The header
+  /// @return          The target threshold
+  extractTarget: (header) => {
+    let d_header = utils.deserializeHex(header)
+
+    // Hacky way of reversing endianness of a partial serialized number
+    let m = header.slice(72, 75).split().reverse().join('')
```

Comment:
This will get characters 72-74 of a hex string. We want bytes 72-74 of a `Uint8Array`.